### PR TITLE
Use JSON11 for handling long numerals

### LINKED ISSUE #6377: [BUG] opensearch-with-long-numerals blocks and times out from Discover page
**Describe the bug**

When attempting to view logs on the Discover page with long-numerals the Kibana instance will fail to respond to requests, causing a failure in health checks as well as returning 500 in Kibana logs.

When attempting the same query from DevTools or outside Kibana (curl) no error is returned and results are supplied.

**To Reproduce**
1. Go to Discover page
2. Create search to include logs with long-numerals
3. Kibana will stop responding for ~5 min.

Kibana logs after ~5min delay of no new logs...
```json
{"type":"log","@timestamp":"2024-04-08T23:14:18Z","tags":["error","opensearch","data"],"pid":1,"message":"[DeserializationError]: Maximum call stack size exceeded"}
{"type":"response","@timestamp":"2024-04-08T23:09:18Z","tags":[],"pid":1,"method":"post","statusCode":500,"req":{"url":"/internal/search/opensearch-with-long-numerals","method":"post","headers":{"x-forwarded-for":"x.x.x.x","x-forwarded-proto":"https","x-forwarded-port":"443","host":"kibana.url.com","x-amzn-trace-id":"Root=1-6614791e-105ca136348f5fd27136bf8f","content-length":"2220","sec-ch-ua":"\"Google Chrome\";v=\"123\", \"Not:A-Brand\";v=\"8\", \"Chromium\";v=\"123\"","content-type":"application/json","osd-xsrf":"osd-fetch","sec-ch-ua-mobile":"?0","user-agent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36","osd-version":"2.13.0","sec-ch-ua-platform":"\"Windows\"","accept":"*/*","origin":"https://kibana.url.com","sec-fetch-site":"same-origin","sec-fetch-mode":"cors","sec-fetch-dest":"empty","referer":"https://kibana.url.com/app/data-explorer/discover","accept-encoding":"gzip, deflate, br, zstd","accept-language":"en-US,en;q=0.9","securitytenant":"tenant"},"remoteAddress":"x.x.x.x","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36","referer":"https://kibana.url.com/app/data-explorer/discover"},"res":{"statusCode":500,"responseTime":300915,"contentLength":9},"message":"POST /internal/search/opensearch-with-long-numerals 500 300915ms - 9.0B"}
```

**Expected behavior**
Kibana is expected to continue responding to other requests while processing.
Kibana is expected to not throw a 500 error.

**OpenSearch Version**
2.13.0

**Dashboards Version**
2.13.0

**Host/Environment (please complete the following information):**
 - OS: CentOS7 from docker image
 - Chrome 123.0.6312.106

### PR DESCRIPTION
**Pull request by AMoo-Miki**
### Description

Replaces the custom regex-based logic with an AST-based one from JSON11. Handling a sample that created 10K false-positives went from 90s to 200ms.

### Issues Resolved
Fixes #6377 

## Changelog
- skip

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -13248,13 +13254,6 @@ minipass@^3.0.0, minipass@^3.1.1:
   dependencies:
     yallist "^4.0.0"
 
-minipass@^4.0.0:
```
Comment (AMoo-Miki) [inline yarn.lock:13251]: This was probably left off of another PR.

File: packages/osd-opensearch/package.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
     "osd:watch": "../../scripts/use_node scripts/build --watch"
   },
   "dependencies": {
-    "@opensearch-project/opensearch": "^2.6.0",
+    "@opensearch-project/opensearch": "^2.9.0",
```
Comment (SuZhou-Joe) [inline packages/osd-opensearch/package.json:15]: Nit: Is it necessary we upgrade the client version?

**Review by SuZhou-Joe — APPROVED** [review]

File: yarn.lock (hunk 1 above)
Comment (kavilla) [reply yarn.lock:13251]: need to rebase? or do you mean like we merged this and it shouldn't be there?

File: packages/osd-std/src/json.ts [hunk 3]
```diff
@@ -3,260 +3,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-/* In JavaScript, a `Number` is a 64-bit floating-point value which can store 16 digits. However, the
```
Comment (kavilla) [inline packages/osd-std/src/json.ts:6]: Huge +1 of the cleanup

**Review by kavilla — APPROVED** [review]
do you think we should run through the closed issues related to the other issues related to long numerals?

thinking we could probably open an issue like 'automate verifying these closed issues?'. i feel like it's particularily perf testing when we see these issues right?

**Comment by kavilla** [comment]
oh weird i see a changelog fragment but i see skip in the pr description.

File: packages/osd-opensearch/package.json (hunk 2 above)
Comment (AMoo-Miki) [reply packages/osd-opensearch/package.json:15]: Yeah. We need the version that uses JSON11

File: yarn.lock (hunk 1 above)
Comment (AMoo-Miki) [reply yarn.lock:13251]: It is a leftover from #6492

**Comment by BionIT** [comment]
Hi @AMoo-Miki This PR has failed backport, could you help to manual backport?

**Pull request merged**